### PR TITLE
Fixing #1192. Falling back to normal way of setting properties when getOwnPropertyDescriptor throws

### PR DESCRIPTION
**Pull request by joeldenning**
See #1192 for an explanation of the bug this fixes. My solution is to be defensive when calling `Object.getOwnPropertyDescriptor`, falling back to just setting the property normally when `getOwnPropertyDescriptor` throws (even though a `setter` is potentially being fired). I didn't dive deep enough into the code to find out how hard it would be to write a test for this use case, but I did test this locally and saw that my app, that was unable to bootstrap in IE11 without this code, is now able to properly bootstrap. Let me know if there's a good way to write a test for it though.

Also, assuming that this is merged, when could I hope for it to be published to npm? The company I work for is hoping to launch a new product in about next two weeks, and I'd rather not have to deploy my fork of systemjs (and probably a forked jspm pointing to my systemjs, too??) to production.

### CODE REVIEW

File: lib/proto.js [hunk 1]
```diff
@@ -63,15 +63,20 @@ function getESModule(exports) {
   if (typeof exports == 'object' || typeof exports == 'function') {
     if (getOwnPropertyDescriptor) {
       var d;
-      for (var p in exports)
-        if (d = Object.getOwnPropertyDescriptor(exports, p))
-          defineProperty(esModule, p, d);
+      for (var p in exports) {
+        try {
```
Comment (guybedford) [inline lib/proto.js]: try - catch deoptimizes the whole `getESModule` function here, so we should probably expand `normalSetProperty` to be a `tryDefineProperty`?

**Comment by guybedford** [comment]
Thanks so much for tracking this down, just let me know if the feedback makes sense?

**Comment by joeldenning** [comment]
I think that I understand what you mean -- just put the try/catch into a separate function so that `getESModule` can still be optimized by V8's compiler (and probably other compilers too). I just added another commit that does that and then retested on my app to ensure that it still fixes the issue and doesn't break anything else.

**Comment by guybedford** [comment]
:+1:

**Pull request merged**